src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+
+
+== Distributed SQL
+
+Distributed SQL allows you to access data stored in the Hazelcast cluster
+declaratively using SQL.
+
+=== Beta Status
+
+The distributed SQL is currently in beta. The compatibility between versions
+is not guaranteed. API might change between versions without notice.
+
+=== How Distributed SQL works
+
+When an SQL statement is submitted for execution, the engine parses and
+optimizes it using Apache Calcite. The result is an executable plan that
+is cached and reused by the subsequent executions of the same statement.
+
+The plan contains a tree of query fragments. A query fragment is a tree of
+operators that could be executed on a single member independently. Child
+fragments supply data to parent fragments, possibly through a network, until
+the root fragment is reached. The root fragment returns query results to a
+user.
+
+Only `IMap` objects are supported at the moment.
+
+=== Querying IMap
+
+==== Names
+
+The engine exposes `IMap` objects as tables in the predefined `partitioned`
+schema using exact names. This schema is in the engine search path so that
+`IMap` objects could be accessed with or without the schema name.
+
+Schema and table names are case sensitive. For example, the `IMap` with the name
+`employee` can be accessed as `employee` or `partitioned.employee`, but not as
+`Employee`.
+
+==== Fields
+
+The engine reads the first local entry of the `IMap` to get the list of fields
+accessible from the SQL. The engine extracts field metadata (name, type)
+from the key and the value, and combines them into a final list of fields.
+
+Field metadata extraction logic depends on the serialization format, as described
+below.
+
+-  For Java objects (`Serializable`, `DataSerializable`, `IdentifiedDataSerializable`),
+the object is deserialized if needed and then analyzed using reflection. Only public
+fields and getters are taken in count. A public field is exposed using its exact name.
+A getter is exposed using the JavaBean naming conventions. E.g. `getName` is exposed as
+`name` and `isActive` is exposed as `active`
+- For `Portable` objects, only the fields that are written in the `Portable.writePortable`
+method are exposed using their exact names
+- `HazelcastJsonValue` objects cannot be queried at the moment. This limitation will be
+removed in future releases
+
+Field names are case-sensitive.
+
+If the `IMap` does not have local entries on the member where the query is started,
+then the list of fields cannot be resolved, and an exception is thrown.
+
+We refer to the key and the value as *top-level* objects. The key is exposed as
```

Comment:
Add examples.

Reply:
Could you please clarify - examples of what?

Reply:
Just in case - I added an example of field resolution to the `Fields` section, which shown how `__key` and `this` are exposed. Chances that it addresses this comment.